Mobile-configuration.md

```diff
@@ -0,0 +1,67 @@
+# Standarize mobile configuration format
+
+## Introduction
+
+Introduce standard for system interoperability, by providing schemas that will be used to communicate between mobile sdk's and core services.
+
+## Problem Description
+
+Mobile configuration format is not standarized. It contains various fields that have no meaning for the mobile users.
+Configuration also exposes some internal fields. 
+
+```json
+{
+ "config": {
+          "headers": {},
+          "mysql_db": "unifiedpush",
+          "mysql_password": "unifiedpush",
+          "mysql_username": "unifiedpush",
+          "name": "unified-push-server",
+          "type": "unified-push-server",
+          "uri": "http://ups-myproject.192.168.37.1.nip.io"
+        },
+        "name": "unified-push-server"
+}
+```
+
+Most important parameters often have different names. For example some services contain `url` property and some `uri`.
+Service specific metadata prevents us from strict json parsing.
+
+## Expectations
+
+- Provide mobile configuration that has only esential fields for mobile users
+- Standarize format so users to not break functionalities once users will start using configuration
+ 
+## Proposed Solution
+
+Provide standard for mobile configuration that will be extendable and have backwards compatibility.
+
+### Mobile cli specifics
+
+Mobile-cli currently print configuration out to the std. 
+Mobile-cli should have option to create `mobile-services.json` file in specified location.
+
+### Json format specific
+
+#### Top level object
+
+```json
+{
+ "version" : "number",
+ "serverUrl" : "string",
+ "namespace": "string",
+ "services": { "type": "array", "items":"Service"}
+}
+```
+
```

Comment:
Might we want to think about client and integration configuration objects in the proposal as well?

I know that on the client side, there are some common configurations we are starting to identify.  We will probably want standardized headers to be attached to all requests, some form of consumable security policy, etc.

For integrations we could probably provide some data about the type of integration we are using.  IE is a service meant to be used as a proxy, do we need to decorate requests, does one service validate requests to another, etc.  (These ideas are all spitballs)

Reply:
I would love to iterate on those and bump versions but for the initial version the main target is to clear some mess and agree on things we know. We can iterate later on schema and bump versions that will allow all SDK's to adjust.

Reply:
Added new info to make this clear.

Reply:
I agree with iteration here. Lets get the first version of the config consumable and then iterate over time. Agreeing here doesn't mean it is what the final GA config will look like.